tt_metal/distributed/mesh_device_view.hpp

```diff
@@ -75,14 +75,13 @@ class MeshDeviceView {
     MeshDeviceView(const MeshDevice& mesh, Coordinate top_left, Coordinate bottom_right);
     MeshDeviceView(std::vector<device_pointer> devices, const CoordinateMapper& mapper);
 
-    [[nodiscard]] device_pointer get_device(size_t row, size_t col);
-    [[nodiscard]] const_device_pointer get_device(size_t row, size_t col) const;
+    [[nodiscard]] device_pointer get_device(size_t row, size_t col) const;
 
     // Get devices spanning the rectangular region defined by the top-left and bottom-right coordinates
     // devices are returned in row-major order with start/end coordinates inclusive
-    [[nodiscard]] DeviceView get_devices(const Coordinate& start, const Coordinate& end);
-    [[nodiscard]] DeviceView get_devices(const MeshShape& shape);
-    [[nodiscard]] DeviceView get_devices(MeshType type = MeshType::RowMajor);
+    [[nodiscard]] DeviceView get_devices(const Coordinate& start, const Coordinate& end) const;
+    [[nodiscard]] DeviceView get_devices(const MeshShape& shape) const;
```

Comment:
`shape` here really means `submesh_shape`, right? Let's rename and clarify the behavior in a comment?

Reply:
I've taken your suggestion to make it clearer.